includes/validation/class-amp-invalid-url-post-type.php

```diff
@@ -1173,258 +1200,114 @@ public static function print_status_meta_box( $post ) {
-									<li>
-										<details <?php echo ! in_array( $key, $collapsed_details, true ) ? 'open' : ''; ?>>
-											<summary><code><?php echo esc_html( $key ); ?></code></summary>
-											<div class="detailed">
-												<?php if ( is_string( $value ) ) : ?>
-													<?php echo esc_html( $value ); ?>
-												<?php else : ?>
-													<pre><?php echo esc_html( wp_json_encode( $value, 128 /* JSON_PRETTY_PRINT */ | 64 /* JSON_UNESCAPED_SLASHES */ ) ); ?></pre>
-												<?php endif; ?>
-											</div>
-										</details>
-									</li>
-								<?php endforeach; ?>
-							</ul>
-						</details>
-					</li>
-				<?php endforeach; ?>
-			</ul>
+		<form class="search-form wp-clearfix" method="get">
+			<input type="hidden" name="taxonomy" value="<?php echo esc_attr( $taxonomy ); ?>" />
+			<input type="hidden" name="post_type" value="<?php echo esc_attr( $post_type ); ?>" />
+			<?php $wp_list_table->search_box( esc_html__( 'Search Errors', 'amp' ), 'invalid-url-search' ); ?>
+		</form>
+		<div id="url-post-filter" class="alignleft actions">
+			<?php AMP_Validation_Error_Taxonomy::render_error_type_filter(); ?>
+			<input name="filter_action" type="submit" id="doaction" class="button action" value="<?php esc_html_e( 'Apply Filter', 'amp' ); ?>">
 		</div>
+
+		<form id="posts-filter" method="post">
+			<input type="hidden" name="taxonomy" value="<?php echo esc_attr( $taxonomy ); ?>" />
+			<input type="hidden" name="post_type" value="<?php echo esc_attr( $post_type ); ?>" />
+			<?php $wp_list_table->display(); ?>
+		</form>
 		<?php
 	}
 
+	/**
+	 * Gets the number of amp_validation_error terms that should appear on the single amp_invalid_url /wp-admin/post.php page.
+	 *
+	 * @todo: consider a way to reduce the number of terms returned in the query, as the pagination is still present.
+	 * @param int $terms_per_page The number of terms on a page.
+	 * @return int The number of terms on the page.
+	 */
+	public static function get_terms_per_page( $terms_per_page ) {
+		global $pagenow;
+		if ( 'post.php' === $pagenow ) {
+			return self::NUMBER_TERMS_ON_SINGLE_PAGE;
+		}
+		return $terms_per_page;
+	}
+
+	/**
+	 * Adds a taxonomy query var to the amp_invalid_url post edit link.
+	 *
+	 * The post.php page that this link leads to is really more like an edit-tags.php page.
+	 * It has a WP_Terms_List_Table of the validation errors (amp_validation_error terms).
+	 * For the WP_Terms_List_Table to render, its $screen->taxonomy property needs to be set.
+	 * So this adds a taxonomy query var, which is later parsed into $screen->taxonomy.
+	 *
+	 * @param string $link    The edit link to filter.
+	 * @param int    $post_id The post ID.
+	 * @param string $context The link context. If set to 'display' then ampersands
+	 *                        are encoded.
+	 * @return string $link The filtered link.
+	 */
+	public static function add_taxonomy_to_edit_link( $link, $post_id, $context ) {
```

Comment:
What do you think of this approach?

This changes the URL of this single URL page from:
https://loc.test/wp-admin/post.php?post=4416&action=edit
to:
https://loc.test/wp-admin/post.php?post=4416&action=edit&taxonomy=amp_validation_error

This is because the `WP_Terms_List_Table` that displays the error terms [needs access to the taxonomy](https://github.com/WordPress/wordpress-develop/blob/fd5ba80c5c3d9cf62348567073945e246285fbca/src/wp-admin/includes/class-wp-terms-list-table.php#L51):
```php
$taxonomy  = $this->screen->taxonomy;
```

The best way I could find so far to make this taxonomy available is by adding it as a query var to that URL, like how it's available in the query var of the taxonomy page (#1361):

https://loc.test/wp-admin/edit-tags.php?taxonomy=amp_validation_error&post_type=amp_invalid_url

It looks like `WP_Screen` [gets that taxonomy](https://github.com/WordPress/wordpress-develop/blob/fd5ba80c5c3d9cf62348567073945e246285fbca/src/wp-admin/includes/class-wp-screen.php#L270) from the query var, and then it [becomes available](https://github.com/WordPress/wordpress-develop/blob/fd5ba80c5c3d9cf62348567073945e246285fbca/src/wp-admin/includes/class-wp-terms-list-table.php#L51) in `WP_Terms_List_Table`.

Reply:
See note above about using `AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG` directly.  You could just do:

```php
$_REQUEST['taxonomy'] = AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG;
```

Just do this before `$wp_list_table = _get_list_table( 'WP_Terms_List_Table' );`.